en/docs/debugging/debug.md

```diff
@@ -0,0 +1,258 @@
+and you are able to execute/debug only the selected test case by clicking on the corresponding codelens as shown below.
+
+![Start_Test Quick Debug Session](/img/start-quick-test-debug-session.gif)
+
+<br/>
+
+### Start a debug session with configurations
+
+Follow the steps below to start a debug session with configurations. All the configurations need to be added in the `launch.json` file.
+
+1. Open the folder, which includes the Ballerina program you want to debug, and select the file.
+
+2. Press the **Control + Shift + D** keys (for Mac: **Command + Shift +D**) to launch the Debugger view.
+
+3. Click **create a launch.json** file and then select **Ballerina Debug** as the **Environment**.
+
+   You view the opened `launch.json` file.
+
+4. Add/edit the relevant configurations for debugging in the `launch.json` file.
+
+5. Add the debug points you require by clicking in front of the line numbers of the file you want to debug.
+
+![Start Debug Session](/img/start-debug-session.gif)
+
+Then, you can start a program, test, or remote debug session as shown below.
+
+>**Info:** If you launch the debug session through VS Code, the working directory will be the Ballerina package root. However, you can use remote debugging for alternative working directories.
+
+#### Start a program debug session with configurations
+
+Follow the steps below to start a program debug session.
+
+1. Select **Ballerina Debug** from the drop-down available in the upper left corner to start a program debugging session.
+
+2. Click the **Start Debugging** icon on the upper left corner to start debugging.
+
+   You view the output in the **DEBUG CONSOLE**.
+
+   ![Program Debug](/img/program-debug.gif)
+
+<br/>
+
+#### Start a test debug session with configurations
+
+Follow the steps below to start a test debug session.
+
+1. Select **Ballerina Test** from the drop-down available in the upper left corner to start a test debugging session.
+
+2. Click the **Start Debugging** icon on the upper left corner to start debugging.
+
+   You view the output in the **DEBUG CONSOLE**.
+
+   ![Test Debug](/img/test-debug.gif)
+
+<br/>
+
+### Start a remote debug session
+
+Follow the steps below to start a remote debug session.
+
+1. Create the `launch.json` configuration file if it is not created already. For instructions on creating the `launch.json` file, see [Start a debug session with configurations](#start-a-debug-session-with-configurations).
+
+2. Open the `launch.json` file and configure the `debuggeeHost` and `debuggeePort` attributes under the `Ballerina Remote` configurations section accordingly.
+
+3. After setting the remote debug configurations, select **Ballerina Remote** from the drop-down available in the upper left corner to start a remote debugging session.
+
+4. Open the Terminal and execute the Ballerina command, which you want to debug out of the supported remote debugging commands below.
```

Comment:
This sentence seems to be out of place.
It implies that we are `debugging a Ballerina command`. Is that correct? 

Is this what we are trying to say?

```
Open a terminal and execute the debug command that is relevant to each issue as explained below.
```

Reply:
Added commit suggestion